core/src/main/java/org/infinispan/configuration/cache/QueryConfiguration.java

```diff
@@ -0,0 +1,34 @@
+package org.infinispan.configuration.cache;
+
+import org.infinispan.commons.configuration.AbstractTypedPropertiesConfiguration;
+import org.infinispan.commons.configuration.attributes.AttributeDefinition;
+import org.infinispan.commons.configuration.attributes.AttributeSet;
+import org.infinispan.commons.configuration.attributes.Matchable;
+
+/**
+ * Configures query options and defaults
+ */
+public class QueryConfiguration extends AbstractTypedPropertiesConfiguration implements Matchable<QueryConfiguration> {
+
+   public static final AttributeDefinition<Integer> DEFAULT_MAX_RESULTS = AttributeDefinition.builder(org.infinispan.configuration.parsing.Attribute.DEFAULT_MAX_RESULTS, 100).immutable().build();
```

Comment:
For now, immutable is fine, but in the future we could turn this into a runtime-mutable attribute. Can you create a Jira for it ?